File: PYTHON/Program/Mobil.py
from Kendaraan import Kendaraan

class Mobil(Kendaraan):
    __jumlahkursi = 4
    __jumlahpintu = 4
    __JenisMobil = ""
    __NamaMobil = ""
    
    def __init__(self, NamaMobil, JenisMobil, jumlahkursi, jumlahpintu, platnomor = "", merk = "", tahun = 2024, warna = "", cckendaraan = 110, tangki = 5):
        super().__init__(platnomor, merk, tahun, warna, cckendaraan, tangki)
        self.NamaMobil = NamaMobil
        self.JenisMobil = JenisMobil
        self.jumlahkursi = jumlahkursi
        self.jumlahpintu = jumlahpintu

    def get_NamaMobil(self):
        return self.NamaMobil

    def set_NamaMobil(self, NamaMobil):
        self.NamaMobil = NamaMobil
    
    def get_JenisMobil(self):
        return self.JenisMobil

    def set_JenisMobil(self, JenisMobil):
        self.JenisMobil = JenisMobil
    
    def get_jumlahkursi(self):
        return self.jumlahkursi

    def set_jumlahkursi(self, jumlahkursi):
        self.jumlahkursi = jumlahkursi

    def get_jumlahpintu(self):
        return self.jumlahpintu

    def set_jumlahpintu(self, jumlahpintu):
        self.jumlahpintu = jumlahpintu
        
    def identifikasi(self):
        identifikasi_kendaraan = super().identifikasi()
        return f"{self.JenisMobil}\n- Nama Mobil :{self.NamaMobil}\n- {identifikasi_kendaraan}\n- Kursi      :{self.jumlahkursi}\n- Pintu      :{self.jumlahpintu}"

File: CPP/Program/Mobil.cpp
#include <iostream>
#include <string>
#include "Kendaraan.cpp"

#pragma once

class Mobil : public Kendaraan {
private:
    std::string JenisMobil;
    std::string NamaMobil;
    int jumlahkursi;
    int jumlahpintu;

public:
    Mobil(std::string NamaMobil, std::string JenisMobil, int jumlahkursi, int jumlahpintu,
          std::string platnomor = "", std::string merk = "", int tahun = 2024, std::string warna = "", int cckendaraan = 110, int tangki = 5) 
        : Kendaraan(platnomor, merk, tahun, warna, cckendaraan, tangki), NamaMobil(NamaMobil), JenisMobil(JenisMobil),
          jumlahkursi(jumlahkursi), jumlahpintu(jumlahpintu) {}

    std::string get_NamaMobil() const {
        return NamaMobil;
    }

    void set_NamaMobil(const std::string& NamaMobil) {
        this->NamaMobil = NamaMobil;
    }

    std::string get_JenisMobil() const {
        return JenisMobil;
    }

    void set_JenisMobil(const std::string& JenisMobil) {
        this->JenisMobil = JenisMobil;
    }

    int get_jumlahkursi() const {
        return jumlahkursi;
    }

    void set_jumlahkursi(int jumlahkursi) {
        this->jumlahkursi = jumlahkursi;
    }

    int get_jumlahpintu() const {
        return jumlahpintu;
    }

    void set_jumlahpintu(int jumlahpintu) {
        this->jumlahpintu = jumlahpintu;
    }

    void identifikasi() const override{
    std::cout << JenisMobil << "\n- Nama Mobil : " << NamaMobil << "\n";
    Kendaraan::identifikasi();
    std::cout << "- Kursi      : " << jumlahkursi << "\n- Pintu      : " << jumlahpintu << std::endl;
}

};